### ISSUE TITLE
Views: Basic settings: improve the label and help text of "Show the embed display in the ui."

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#1461]
A few things wrong here:

- "ui" should be capitalized.
- should be "Allow advanced user**s** to use the embed view display. The plugin itself works if it's not visible in the **UI**" (plural for "users" and again capitalized "UI")

That is purely from a typographical point of view. In general though... WTF??? I don't get it and it resides in the "basic" settings. Even its description talks about "advanced users".

**Comment by klonos** [comment]
...it turns out that there's an issue for than against D8: https://www.drupal.org/node/1837998

The label and text have changed to this:

> [ ] Allow embedded displays
> *Embedded displays can be used in code via views_embed_view().*

They seem to have removed a lot of the help text and trimmed the checkbox labels too. Here's what the "Basic settings" page looks like in Backdrop:

![backdrop-views_ui-basic_settings](https://cloud.githubusercontent.com/assets/2423362/12037665/35160c04-aea5-11e5-8088-41f4b34c3812.png)

...and in D8:

![d8-views_ui-basic_settings](https://cloud.githubusercontent.com/assets/2423362/12037668/3c15c5da-aea5-11e5-9604-010ccf7e4247.png)

**Comment by klonos** [comment]
...not sure whether "Show in the UI" or "Allow" is better? Does it mean that if this checkbox is not checked in D8, then the use of embedded views is not allowed whatsoever?

Also, what about the other settings available in the "Live preview settings" section? ...are they things implemented in Views after we merged or specifically developed in the version bundled with D8 core? ...should we implement these too for the sake of feature parity? Which begs the question: how do we follow the developments in Views in general (both the 7.x contrib as well as the D8 core one)?

**Comment by quicksketch** [comment]
> Also, what about the other settings available in the "Live preview settings" section?

These are all in Backdrop, you just have to check the "Show information and statistics" checkbox for them to show up.

![selection_015](https://cloud.githubusercontent.com/assets/100206/12059674/a5c19a08-af14-11e5-861b-c0cb99403046.png)

> Which begs the question: how do we follow the developments in Views in general (both the 7.x contrib as well as the D8 core one)?

D7 development has cooled down, but we can monitor new releases like we do for D7 releases. D8 changes... not sure maybe a la carte?

In addition to everything you've listed here. I'd also be in favor of removing the "Custom admin theme for the Views UI" option.

And I think we should consolidate the "Advanced" settings with the "Basic" ones. As you point out, the basic ones aren't really very basic. The ability to clear the views cache only is obscure and can be done via Admin Bar anyway. The "Debugging" options can all just go in a collapsed fieldset IMO.

**Comment by klonos** [comment]
Created a PR for this...

Before:

> [ ] Always show the master display
>   Advanced users of views may choose to see the master (i.e. default) display.
>
> [ ] Show the embed display in the ui.
>   Allow advanced user to use the embed view display. The plugin itself works if it\'s not visible in the ui


After:

> [ ] Always show the _Master_ display
>   Show the _Master_ (i.e. default) display, which is otherwise hidden by default.
>
> [ ] Allow _embed_ displays to be created via the UI
>   This enables another type of views display called _embed_, which can be added via the "+ Add" button when creating/editing a view. This type of display can be used in code via [`views_embed_view()`](https://api.backdropcms.org/api/backdrop/core%21modules%21views%21views.module/function/views_embed_view). The plugin for _embed_ displays itself works even if this setting is disabled.

**Comment by klonos** [comment]
...screenshot:

![screen shot 2018-11-22 at 10 57 07 pm](https://user-images.githubusercontent.com/2423362/48901593-0628fa80-eeaa-11e8-9169-04016767c81d.png)

**Comment by docwilmot** [comment]
Second one too verbose

**Comment by olafgrabienski** [comment]
Saw a typo in the screenshot:

> The plugin for ***emebed*** displays itself works even if ...

**Comment by klonos** [comment]
> Saw a typo in the screenshot

Yep, the screenshot was before I actually filed the PR. I saw that and fixed it 😄 

> Second one too verbose

I know, but it was the very minimal I could come up with that would communicate the purpose of that setting without leaving room for doubt or misinterpretation. Suggestions please?

**Comment by klonos** [comment]
...how about this:

> Enables the _embed_ type of views display, which can be used in code via [`views_embed_view()`](https://api.backdropcms.org/api/backdrop/core%21modules%21views%21views.module/function/views_embed_view). The plugin for _embed_ displays itself works even if this setting is disabled.

**Comment by docwilmot** [comment]
I don't particularly like the mentioning a particular function in the description. I prefer:
> "Allow users to use the embed view display, which can be used in code. The display plugin itself works if it's not visible in the UI"

**Comment by klonos** [comment]
@docwilmot:

1. I prefer "Enables the _embed_ type of views display ..." because it clearly tells people that a new option for displays will be added in the UI.

2. "The display plugin itself works if it's not visible in the UI." makes it sound like the display plugin is visible in the UI; whereas it is the display that is enabled.

3. >I don't particularly like the mentioning a particular function in the description.

Fair enough.

So how about this?:

> Enables the _embed_ type of views display, which can be used in code. The plugin itself works even if the display is not visible in the UI.

**Comment by klonos** [comment]
Good point @docwilmot 👍...here's another take:

> Adds _embed_ as an available display type in the views UI. _Embed_ displays can be used in code, and that will work even if the display is not visible in the UI.

**Comment by bugfolder** [comment]
Given the age, the PR https://github.com/backdrop/backdrop/pull/2390 probably needs rebasing, plus whatever magic is needed to invoke the new automated tests.
